src/runtime-rs/crates/resource/src/manager_inner.rs

```diff
@@ -198,3 +202,48 @@ impl ResourceManagerInner {
         self.volume_resource.dump().await;
     }
 }
+
+#[async_trait]
+impl Persist for ResourceManagerInner {
+    type State = ResourceState;
+    type ConstructorArgs = ManagerArgs;
+    /// Save a state of the component.
+    async fn save(&self) -> Result<Self::State> {
+        let mut endpoint_state = vec![];
+        if let Some(network) = &self.network {
+            if let Some(ens) = network.save().await {
+                endpoint_state = ens;
+            }
+        }
+        let cgroup_state = self.cgroups_resource.save().await?;
+        Ok(ResourceState {
+            endpoint: endpoint_state,
+            cgroup_state: Some(cgroup_state),
+        })
+    }
+    /// Restore a component from a specified state.
+    async fn restore(
+        resource_args: Self::ConstructorArgs,
+        resource_state: Self::State,
+    ) -> Result<Self> {
+        let args = CgroupArgs {
+            sid: resource_args.sid.clone(),
+            config: resource_args.config,
+        };
+        Ok(Self {
+            sid: resource_args.sid,
+            agent: resource_args.agent,
+            hypervisor: resource_args.hypervisor,
+            network: None,
+            share_fs: None,
+            rootfs_resource: RootFsResource::new(),
+            volume_resource: VolumeResource::new(),
```

Comment:
It's a shame that not all the new types implement Default as the code could be simplified if they did.

Reply:
As the functionality of cleanup is not fully support at this stage, this is pr is aiming to give a framework for that. The problem you mentioned would be solved in the future.